src/Listener/PushNewPost.php

```diff
@@ -28,10 +28,12 @@ public function __construct(Pusher $pusher)
     public function handle(Posted $event)
     {
         if ($event->post->isVisibleTo(new Guest)) {
+            $tags = $event->post->discussion->tags;
+
             $this->pusher->trigger('public', 'newPost', [
                 'postId' => $event->post->id,
                 'discussionId' => $event->post->discussion->id,
-                'tagIds' => $event->post->discussion->tags()->pluck('id')
+                'tagIds' => $tags ? $tags->pluck('id') : null
```

Comment:
Isn't tags being changed from a function call to an attribute call? I'm not too familiar with Eloquent's magic, but wouldn't this change behavior?

Reply:
If I remember correctly this should still work because the tags attribute call will returns a collection.

Reply:
It will still work, but what happens behind the scenes is different.

If I'm not wrong, calling it as a function returns a query, from which you select only the `id` column (SQL). Not calling it as a function means all the columns are queried from the DB, and `pluck` just converts that collection with all the attributes into just IDs. In other words, all columns are now queried from DB, which could increase performance hits.

Reply:
What Laravel does in the Model/Eloquent class is this for attributes:

- see if there's an accessor, use that
- see if there's a relation, load the relation and provide the Model or Collection
- see if there's a property/attribute, use that

In case of a function call it would hit the relationship method. Which needs to be defined. By using the property it will return `null` if no accessor, relation or property exists. Very helpful in this case.

The benefit of using the method call is that it **always** circumvents any cached relationship results from the model. However, when saving/changing relationship values from inside the model these values are usually immediately updated.

I therefor see no issue in using the direct property in this piece of code.

Reply:
Thanks!